### ISSUE TITLE
 Some questions about the WISE

### ISSUE THREAD

**Issue by sev777** [zjunlp/EasyEdit#423]
I run wise on zsre with Llama-3-8b-instruct in single Editing case, but the results didn't change befor and after editng.  

I get the results:
`Metrics Summary:  {'pre': {'rewrite_acc': 0.29999999999999993, 'rephrase_acc': 0.29999999999999993}, 'post': {'rewrite_acc': 0.29999999999999993, 'rephrase_acc': 0.29999999999999993, 'locality': {'neighborhood_acc': 1.0}}}`

And why  `act_mask  `and `deact_mask  `is always None in wise\utils.py ?
I find after edit single data. the layer_out won't use the new_weight in WISE.py.
   
```
       if min_dist.item() < threshold:
                    print(dist1.item(), threshold, 'ori')
                    layer_out = original_layer_output
       else:
                    print(dist1.item(), threshold, 'new')
                    layer_out = new_weight_layer_output
```
And if  I print the dist1.item() and  threshold, it looks like:
```
rephrase:
22.316946029663086 25.004183654785155 ori
22.84709930419922 25.004183654785155 ori
locality:
10.457975387573242 25.004183654785155 ori
```

The setting I use is same as the default:

```
mask_ratio: 0.2
edit_lr: 1.0
n_iter: 70
norm_constraint: 1.0
act_margin: [5, 20.0, 10.0] # alpha, beta, gamma  5-》1
act_ratio: 0.88
save_freq: 500
merge_freq: 1000
merge_alg: 'ties'
objective_optimization: 'only_label'
inner_params:
- model.layers[27].mlp.down_proj.weight


## alternative: WISE-Merge, WISE-Retrieve

# for merge (if merge)
densities: 0.53
weights: 1.0

# for retrieve (if retrieve, pls set to True)
retrieve: True
replay: False # True --> will replay the past editing instances: see https://arxiv.org/abs/2405.14768 Appendix B.3

model_parallel: False
use_chat_template: True

```

**Comment by pengzju** [comment]
You can set **act_ratio to 0.7.** The original paper tested llama-2, and the configuration of llama-3 may need slight adjustments.



> And why act_mask  and deact_mask  is always None in wise\utils.py ?

You can ignore this. `act_mask` is used to block the activation of some tokens. In the final version, it is always `None`.

**Comment by sev777** [comment]
> You can set **act_ratio to 0.7.** The original paper tested llama-2, and the configuration of llama-3 may need slight adjustments.
> 
> > And why act_mask  and deact_mask  is always None in wise\utils.py ?
> 
> You can ignore this. `act_mask` is used to block the activation of some tokens. In the final version, it is always `None`.

Thanks, I try  0.5 is work, but 0.7 couldn't. 

Does `act_mask` used to mask the mask tokens for the input? because the length of each input is different.

**Comment by pengzju** [comment]
Thanks a lot. `act_mask` represents the threshold for activation classification. **It takes the norm of all tokens for each input**, but it's just scaled numerically through act_mask.

**Comment by sev777** [comment]
```
    if act_mask is not None:
        return torch.sum(l2_norm * act_mask, dim=1) / torch.sum(act_mask, dim=1)
```
Suppost the shape of l2_norm is 5 * 30,  and the length for 5 data is [30,28,26,15,19]. Then the act_mask could used to mask the MASK token in other 4 data which the length is [28,26,15,19]. 
Is it right?

**Comment by zxlzr** [comment]
hi, do you have any further issues?
